### PR TITLE
Add email field to Contact Info Widget

### LINKED ISSUE #3293: Contact Info widget: Add email address field
We should add a field to the Contact Info widget for an email address. It should auto-format the email entered to be a clickable mailto link.

There likely are a couple of reasons we don't include an email address field on the Contact Form widget, possibly for either spam reasons, or perhaps because the widget is used for brick and mortar businesses. But for users it feels like an oversight. There are many reasons a site owner would choose to include an email address in their contact information, along with their address and phone number. For example, a restaurant could add the email address of their reservation system.

Issue raised in 2495315-t

### PR DESCRIPTION
**Pull request by Stoyan0v**
Fixes #3293

#### Changes proposed in this Pull Request:

* Add an email field to Contact Info Widget

#### Testing instructions:

* Add a Contact Info widget to any sidebar
* Make sure the `admin_email` is set as default email address
* Make sure that the same email is displayed in the front end and its link contains `mailto:`

#### Additional Notes:
I am not sure if we need to validate the email before we display it. For example, we can replace: 

`if ( '' != $instance['email'] ) {`

with: 

`if ( '' != $instance['email'] && is_email( $instance['email'] ) ) {`

### CODE REVIEW

File: modules/widgets/contact-info.php [hunk 1]
```diff
@@ -59,6 +59,7 @@ public function defaults() {
 				'address' => __( "3999 Mission Boulevard,\nSan Diego CA 92109", 'jetpack' ),
 				'phone'   => _x( '1-202-555-1212', 'Example of a phone number', 'jetpack' ),
 				'hours'   => __( "Lunch: 11am - 2pm \nDinner: M-Th 5pm - 11pm, Fri-Sat:5pm - 1am", 'jetpack' ),
+				'email'   => get_bloginfo( 'admin_email' ),
```
Comment (samhotchkiss) [inline modules/widgets/contact-info.php]: I'm not crazy about using the admin email as the default.  Why not just do something like `jane.doe@gmail.com`?

File: modules/widgets/contact-info.php [hunk 2]
```diff
@@ -123,6 +124,10 @@ function widget( $args, $instance ) {
 				}
 			}
 
+			if ( '' != $instance['email'] ) {
```
Comment (samhotchkiss) [inline modules/widgets/contact-info.php]: Let's check if `is_email` before linking-- if it's not a valid email, show it as plain text, but not clickable

Comment (Stoyan0v) [inline modules/widgets/contact-info.php]: I've added additional check and if the email is invalid, it will be displayed as plain text.

File: modules/widgets/contact-info.php (hunk 1 above)
Comment (Stoyan0v) [inline modules/widgets/contact-info.php]: I agree, but instead `gmail.com` I prefer `example.com`

File: modules/widgets/contact-info.php (hunk 2 above)
Comment (jeherve) [inline modules/widgets/contact-info.php]: If the email is invalid, maybe we shouldn't be displaying anything? What's the advantage of displaying an invalid email here?

This would also simplify the whole thing, I think:

```php
if ( is_email( $instance['email'] ) ) {
	printf(
		'<div class="confit-email"><a href="mailto:%s">%s</a></div>',
		esc_html( $instance['email'] )
	);
}
```

Comment (Stoyan0v) [inline modules/widgets/contact-info.php]: Imo, the reason that we should display the text if it's not an email, is that you can add multiple email addresses or some text before or after the email.

Comment (jeherve) [inline modules/widgets/contact-info.php]: If we want to accept additional text or multiple email addresses, maybe we should make that a new field then, as well as modify the new Email address field to specify that one can use multiple email addresses.

That said, I'm not sure it will benefit many people. One single email address seems like a good standard to me.

Comment (Stoyan0v) [inline modules/widgets/contact-info.php]: Ok, then we will display the email if it's valid only.

**Comment by dereksmart** [comment]
This adds the field and default value automatically, so when I checked out to this branch the Widget I already had configured had an email field added with the example address displayed on the front end.  

tl;dr this would add dummy email address for anyone who upgraded.  

Let's default to not display anything, and to hide the field if not in use.

**Comment by Stoyan0v** [comment]
@dereksmart, just to confirm: 
* The field will don't have default value
* If the email is invalid, it will not be displayed 

Also I am not sure what does `hide the field if not in use` means. 
Do you mean to hide the field in backend?

**Comment by dereksmart** [comment]
@Stoyan0v Nope sorry, just the front end.  Just want to make sure no one sees any surprises on the front end of their site after upgrade :)

**Comment by Stoyan0v** [comment]
I have addressed the changes. To sum up: 
* The field will don't have default value
* If the email is invalid, it won't be displayed.

**Comment by singerb** [comment]
Ran a quick test, looks good to me, and the changes above seem to be satisfied.

**Comment by dereksmart** [comment]
Looks good thanks @Stoyan0v 👍
